dev/versioning.md

```diff
@@ -0,0 +1,52 @@
+# Semantic Versioning Policy
+
+`ember-template-lint` follows [semantic versioning](http://semver.org/) and [ESLint's Semantic Versioning Policy](https://github.com/eslint/eslint#semantic-versioning-policy).
+
+For clarity, we define semver policy for this addon in two parts- as related to **rules** and as related to **config** files. 
+
+## Related to Rules 
+
+### Patch Release
+
+A patch release should not break your lint build.
+
+* A bug fix in a rule that results in it reporting fewer errors.
+* Improvements to documentation.
+* Non-user-facing changes such as refactoring code, adding, deleting, or modifying tests, and increasing test coverage.
+* Re-releasing after a failed release (i.e., publishing a release that doesn't work for anyone).
+
+### Minor Release
+
+A minor release may break your lint build in some cases.
+
+* A bug fix in a rule that results in it reporting more errors.
+* A new rule is created.
+* A new option to an existing rule is created.
+* An existing rule is deprecated.
+
+### Major Release
+
+A major release is likely to break your lint build.
+
+* Support for an old Node version is dropped.
+* An existing rule is changed in it reporting more errors.
+* An existing rule is removed.
+* An existing option of a rule is removed.
+
+## Config Files
+
+For dynamically generated config files such as `a11y-strict` and `all`, the config will change as needed for any release and therefore cannot be covered by a semver guarantee.
```

Comment:
I think this part _could_ be added when we land the first of these (over
#946) if we want to punt on this part for now.